src/serializers/encoder/v12/encode_schema.c

```diff
@@ -76,23 +76,11 @@ static inline void _RdbSaveExactMatchIndex
 
 	uint fields_count = Index_FieldsCount(idx);
 
-	// for exact-match index on an edge type, decrease `fields_count` by 2
-	// skipping `_src_id` and `_dest_id` fields
-	uint encode_fields_count =
-		type == SCHEMA_EDGE ? fields_count - 2 : fields_count;
-
 	// encode field count
-	RedisModule_SaveUnsigned(rdb, encode_fields_count);
+	RedisModule_SaveUnsigned(rdb, fields_count);
 	for(uint i = 0; i < fields_count; i++) {
 		char *field_name = idx->fields[i].name;
 
-		// for exact-match index on an edge type, skip both `_src_id` and
-		// `_dest_id` fields, these are introduce automaticly by the index
-		// construct routine
-		if(type == SCHEMA_EDGE && 
```

Comment:
on older ribs that have this fields it is problematic

Reply:
I don't follow, please explain